gatsby-theme-landing-page/src/components/markdown-text.js

```diff
@@ -21,14 +21,26 @@ const sanitizeOptions = {
   selfClosing: ["img", "hr"],
 };
 
-export default function MarkdownText({ childMarkdownRemark }) {
+const inlineOptions = {
+  allowedTags: ["strong", "b", "i", "em", "a", "span"],
+  selfClosing: [],
+};
+
+export default function MarkdownText({
+  childMarkdownRemark,
+  as = "div",
```

Comment:
I like this API, some suggestions:
- We check `as` against a list of accepted elements, e.g. [div, h1...h6]
- We automatically apply the `inline` options against heading tags
- The upshot here is that consumers generally don't need to think about whether something "should" be inline or not; we display headings correctly when that's the provided argument, and otherwise we assume a div with the full set of accepted tags.

Reply:
Yeah, can make this more automagic and was debating something similar. It might make sense to leave it rendering as a `div` + all block elements when `as` is undefined, then if the `as` prop is defined, render it with only inline elements.

As for an allowlist for this, I think it's better to keep the API open since it could be frustrating to a developer using this component if certain values just failed silently and I can't predict how we might want to use this for other components yet